code/game/turfs/simulated/floor_attackby.dm

```diff
@@ -173,11 +173,15 @@
 	update_icon()
 	return 1
 
+/turf/simulated/floor/try_build_cable(obj/item/stack/cable_coil/C, mob/user)
+	if(!can_build_cable(user))
+		if(!is_plating() || flooring)
+			to_chat(user, SPAN_WARNING("Removing the tiling first."))
+		if(broken || burnt)
+			to_chat(user, SPAN_WARNING("This section is too damaged to support anything. Use a welder to fix the damage."))
+		return FALSE
```

Comment:
I've found the superior pattern is to have the check proc return either an error string, possibly an unspecified error signal (like TRUE), or a success (like FALSE or null). The issue with the approach here is that I can potentially override to have a turf not require e.g. !burnt for cabling, but it's then not clear how to override this proc without duplicating code.

Not sure others agree with me here; I haven't seen anyone else use that.

Reply:
I don't think fixing it by making some complex return code stuff is really the best way here. This should be very simple and concise.
And its more likely that if they changed things so much as to need to change those pretty basic conditions they'd probably need to make a new proc to handle that on the cable coil. And then they'd need to bypass the parent proc to call their own anyways.